mmv1/third_party/terraform/acctest/bootstrap_test_utils.go

```diff
@@ -1327,3 +1328,144 @@ func SetupProjectsAndGetAccessToken(org, billing, pid, service string, config *t
 
 	return accessToken, nil
 }
+
+func BootstrapVmwareengineNetwork(t *testing.T, networkName string) string {
+	project := envvar.GetTestProjectFromEnv()
+
+	config := BootstrapConfig(t)
+	if config == nil {
+		return ""
+	}
+	log.Printf("[DEBUG] Getting Vmware Engine Network %q", networkName)
+	url := fmt.Sprintf("%sprojects/%s/locations/global/vmwareEngineNetworks/%s", config.VmwareengineBasePath, project, networkName)
+	_, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+		Config:               config,
+		Method:               "GET",
+		Project:              project,
+		RawURL:               url,
+		UserAgent:            config.UserAgent,
+		ErrorAbortPredicates: []transport_tpg.RetryErrorPredicateFunc{transport_tpg.Is429QuotaError},
+	})
+	if err != nil {
+		log.Printf("[DEBUG] Vmware Engine Network %q not found, bootstrapping", networkName)
+		networkObj := map[string]interface{}{
+			"type": "STANDARD",
+		}
+		networkCreateUrl := fmt.Sprintf("%sprojects/%s/locations/global/vmwareEngineNetworks?vmwareEngineNetworkId=%s", config.VmwareengineBasePath, project, networkName)
+		networkRes, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+			Config:               config,
+			Method:               "POST",
+			Project:              project,
+			RawURL:               networkCreateUrl,
+			UserAgent:            config.UserAgent,
+			Body:                 networkObj,
+			Timeout:              20 * time.Minute,
+			ErrorAbortPredicates: []transport_tpg.RetryErrorPredicateFunc{transport_tpg.Is429QuotaError},
+		})
+		if err != nil {
+			t.Fatalf("Error bootstrapping Vmware Engine Network %q: %s", networkName, err)
+		}
+
+		log.Printf("[DEBUG] Waiting for Vmware Engine Network creation to finish")
+		err = vmwareengine.VmwareengineOperationWaitTime(
+			config, networkRes, project, "Creating Vmware Engine Network", config.UserAgent,
+			20*time.Minute)
+		if err != nil {
+			t.Errorf("Error getting Vmware Engine Network %q: %s", networkName, err)
+		}
+
+		_, err = transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+			Config:               config,
+			Method:               "GET",
+			Project:              project,
+			RawURL:               url,
+			UserAgent:            config.UserAgent,
+			ErrorAbortPredicates: []transport_tpg.RetryErrorPredicateFunc{transport_tpg.Is429QuotaError},
+		})
+		if err != nil {
+			t.Errorf("Error getting shared Vmware Engine Network %q: %s", networkName, err)
+		}
+	}
+
+	networkId := fmt.Sprintf("projects/%s/locations/global/vmwareEngineNetworks/%s", project, networkName)
+	return networkId
+}
+
+func BootstrapVmwareenginePrivateCloud(t *testing.T, privateCloudName string, networkName string) string {
```

Comment:
I'm a bit worried about this. Judging by the [pricing page](https://cloud.google.com/vmware-engine/pricing) this resource is very expensive to run (~$9/hr), and this bootstrap function will create one and leave it around. That could cause unexpectedly large bills for users who run this test

Reply:
I think we need to figure out another solution here rather than leaving a private cloud around in any project we run this test in.

Can we add this datasource test as part of the existing private cloud test? It's unusual, but I think that due to the extremely long time to deploy it could be a good solution.

That would look like adding the datasource to the existing test's config and adding the testAccCheckGoogleVmwareengineNsxCredentialsMeta to that test. You can add a comment in there about why it exists because it would be an unusual structure